sympy/integrals/transforms.py

```diff
@@ -67,6 +67,18 @@ def __init__(self, transform, function, msg):
         self.function = function
 
 
+def _tree(f, name, **hints):
+    """
+    Helper function creating debug messages that will show the recursion
+    tree if `sympy.SYMPY_DEBUG=True` the stderr is subjected to
+    `grep '[tree]'`.
+    """
+    r = hints.get('recursive', 0)
+    s = '[tree] ' + '  '*r + '|-- ' + name + ': %s'%(f, )
+    debug(s)
```

Comment:
It would be great if this could avoid computing anything if it isn't actually going to output anything. For example formatting `f` as a string is a potentially expensive operation.

Reply:
This happens all over the codebase, e.g., also in meijerint.  Would it be OK to write a second `debug` routine in utils that takes a string and a tuple and only does the formatting `string%tuple` if SYMPY_DEBUG is true? Maybe called `debugf` after printf? Or should I do that locally in `transforms.py`?

Reply:
Yes, I think what we really want is something that works more like Python's logging module which doesn't format the message if it won't be printed. So you'd do something like `debug('value is %r', myvalue)`.

Reply:
`debug()` is not a public function. I think it would be fine to change the signature to `debug(msg, *fmtargs, **fmtkwargs)`.